lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
@mikz , 

> Shouldn't this use just #foreign_key method? 

for doing this I should change all the doubles to allow foreign_key method, do you want this? 

>> Because I guess we don't want to show fields for associations with no options. 

in this case options will be empty hash and ```compact``` will ignore such columns . Also  we do care only for non-standard foreign keys which are described using ```:foreing_key```  option. 

>>Also imo it should check just ActiveRecord::Reflection::BelongsToReflection as the other foreign keys will not exist on this table. That should be easily checked with Reflection#belongs_to?

is this is ok for you ?

```ruby
 def foreign_key_columns
     if @object.present? && @object.class.respond_to?(:reflections)
         @object.class.reflections.select{|_, v| v.macro == :belongs_to }.map {|_,v|  v.options[:foreign_key] }.compact
     else
        []
     end
   end
```